pilot/pkg/model/gateway.go

```diff
@@ -268,9 +269,18 @@ func MergeGateways(gateways []gatewayWithInstances, proxy *Proxy, ps *PushContex
 							}
 							serversByRouteName[routeName] = []*networking.Server{s}
 						}
-
+						if bindsMap, ok := nonPlainTextGatewayPortsBindMap[resolvedPort]; ok {
+							bindsMap[serverPort.Bind] = "true"
```

Comment:
if the map is only going to container "true" it should just be a set. `map[uint32]sets.String`

Reply:
Updated to map[uint32]sets.String, thanks!